tarantool/connection.py

```diff
@@ -897,10 +907,35 @@ def connect_basic(self):
         :meta private:
         """
 
-        if self.host is None:
-            self.connect_unix()
-        else:
+        if self.socket_fd is not None:
+            self.connect_socket_fd()
+        elif self.host is not None:
             self.connect_tcp()
+        else:
+            self.connect_unix()
+
+    def connect_socket_fd(self):
+        """
+        Establish a connection using an existing socket fd.
+
+        socket_fd / timeout|         >= 0            |         None
+        -------------------|-------------------------|-------------------------
+              blocking     | set non-blocking        |  don't change
+                           | socket lib call `select`| `select` isn't needed
+        -------------------|-------------------------|-------------------------
+            non-blocking   | don't change            |  don't change
+                           | socket lib call `select`|  call `select` ourselves
+        -------------------|-------------------------|-------------------------
```

Comment:
Current table formatting is not good for sphinx. You can test it by yourself by removing `:meta private:` and building the docs. Use this approach instead. (AFAIR, additional indentation is also required.)

```suggestion
            +---------------------+--------------------------+-------------------------+
            | socket_fd / timeout |         >= 0             |         `None`          |
            +=====================+==========================+=========================+
            |      blocking       | Set non-blocking socket  | Don't change, `select`  |
            |                     | lib call `select`        | isn't needed            |
            +---------------------+--------------------------+-------------------------+
            |     non-blocking    | Don't change, socket lib | Don't change, call      |
            |                     | call `select`            | `select` ourselves      |
            +---------------------+--------------------------+-------------------------+
```

Reply:
Fixed.